src/index.ts

```diff
@@ -1470,13 +1465,9 @@ function FlatpickrInstance(
             ~(e as any).path.indexOf(self.altInput)));
 
       const lostFocus =
-        e.type === "blur"
-          ? isInput &&
-            e.relatedTarget &&
-            !isCalendarElem(e.relatedTarget as HTMLElement)
-          : !isInput &&
-            !isCalendarElement &&
-            !isCalendarElem(e.relatedTarget as HTMLElement);
+        !isInput &&
+        !isCalendarElement &&
+        !isCalendarElem(e.relatedTarget as HTMLElement);
```

Comment:
Removing `e.type === "blur"` since it's confusing.
We used to accept blur events in the `documentClick()`, but we no longer after https://github.com/flatpickr/flatpickr/commit/a7acd8fbc59c65fcfb730d30e6abc9e79b1bdf8e.